include/deal.II/numerics/coarsening_strategies.h

```diff
@@ -0,0 +1,94 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_coarsening_strategies_h
+#define dealii_coarsening_strategies_h
+
+
+#include <deal.II/base/config.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * When data is transferred during coarsening, it is not trivial to decide
+ * how to handle data of child cells which will be coarsened. Or in other
+ * words, which data should be stored in the corresponding parent cell.
+ *
+ * In this namespace, we offer a few strategies that cope with this
+ * problem. Such strategies can be passed to the CellDataTransfer and
+ * parallel::distributed::CellDataTransfer constructors.
+ *
+ * @author Marc Fehling, 2019
+ */
+template <typename value_type>
+struct CoarseningStrategies
+{
+  /**
+   * Check if data on all children match, and return that value.
+   */
+  static value_type
+  check_equality(const std::vector<value_type> &children_values)
+  {
+    Assert(children_values.size() > 0, ExcInternalError());
+    const value_type value = children_values[0];
+
+    for (unsigned int index = 1; index < children_values.size(); ++index)
+      Assert(value == children_values[index],
+             ExcMessage(
+               "Values on cells that will be coarsened are not equal!"));
+
+    return value;
+  }
```

Comment:
It doesn't make a big difference, but we usually separate function definitions and declarations (even if they are in the same file). We should do this.